runtime/adapters/src/price.rs

```diff
@@ -19,3 +28,76 @@ where
 		OP::price(&route, P::get())
 	}
 }
+
+pub struct FeeAssetBalanceInCurrencyConvertor<T, P, Period, AC, I>(sp_std::marker::PhantomData<(T, P, Period, AC, I)>);
+
+impl<T, P, Period, AC, I> FeePaymentCurrencyBalanceInCurrency<AssetId, T::AccountId>
+	for FeeAssetBalanceInCurrencyConvertor<T, P, Period, AC, I>
+where
+	T: pallet_ema_oracle::Config + frame_system::Config,
+	P: PriceOracle<AssetId, Price = EmaPrice>,
+	Period: Get<OraclePeriod>,
+	AC: FeePaymentCurrency<T::AccountId, AssetId = AssetId>,
+	I: frame_support::traits::fungibles::Inspect<T::AccountId, AssetId = AssetId, Balance = Balance>,
+{
+	type Output = (Balance, Weight);
+
+	fn get_balance_in_currency(to_currency: AssetId, account: &T::AccountId) -> Self::Output {
+		let Some(from_currency) = AC::get(account) else {
+			return (0,Weight::zero());
+		};
+		let account_balance = I::reducible_balance(from_currency, account, Preservation::Preserve, Fortitude::Polite);
+
+		if from_currency == to_currency {
+			return (account_balance, T::DbWeight::get().reads(2));
+		}
+		// We get the weight from the ema-oracle weights to get price
+		// Weight * 2 because we are reading from the storage twice ( from_currency/lrna and lrna/to_currency)
+		// TODO: it could really be a part of the PriceOracle trait?!
+		// if this gets removed, the constraint on T and ema-oracle is not necessary
+		let price_weight = pallet_ema_oracle::Pallet::<T>::get_price_weight()
+			.saturating_mul(2)
+			.saturating_add(T::DbWeight::get().reads(2));
+		let Some(price) = P::price(&[Trade {
```

Comment:
what if we have a fee asset in other pool like stablepool? If we want so, we could get the route from onchain.

The same is true for `ConvertToCurrencyUsingOracle`

Reply:
why are we always writing new price adapter, this should be resolved 10 times already

Reply:
i will change to this to use the exiting price with route. makes sense